pkg/cloud/gcp/actuators/machine/machine_scope.go

```diff
@@ -90,9 +93,12 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 	gcpLabelsTagsFeature := false
 	if params.featureGates != nil {
 		gcpLabelsTagsFeature = params.featureGates.Enabled(configv1.FeatureGateGCPLabelsTags)
-		klog.V(1).Infof("status of %v feature is %t", configv1.FeatureGateGCPLabelsTags, gcpLabelsTagsFeature)
+		klog.V(3).Infof("status of %v feature is %t", configv1.FeatureGateGCPLabelsTags, gcpLabelsTagsFeature)
 	}
 
+	tagManager := util.NewTagManager(params.coreClient, util.NewTagValuesClient, serviceAccountJSON,
```

Comment:
Does this function return nil if the feature is disabled? I would prefer if the switch logic (should we use this feature or not) were at the call site, rather than within the tags manager itself. It makes it hard to reason about when debugging.

The tag manager should be the production ready code, it shouldn't need to carry or know about the state of the feature.

Reply:
I will update to have the featuregate check at caller to invoke TagManager only when enabled.

Reply:
suggestion incorporated.